src/legacy/ui/public/vis/editors/default/components/sidebar/state/actions.ts

```diff
@@ -0,0 +1,171 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { AggConfig, Vis, VisParams } from 'ui/vis';
+import { EditorStateActionTypes } from './constants';
+import { Schema } from '../../../schemas';
+
+export interface ActionType<T, P> {
+  type: T;
+  payload: P;
+}
+
+type AggId = AggConfig['id'];
+type AggParams = AggConfig['params'];
+
+type AddNewAgg = ActionType<EditorStateActionTypes.ADD_NEW_AGG, { schema: Schema }>;
+type DiscardChanges = ActionType<EditorStateActionTypes.DISCARD_CHANGES, Vis>;
+type ChangeAggType = ActionType<
+  EditorStateActionTypes.CHANGE_AGG_TYPE,
+  { aggId: AggId; value: AggConfig['type'] }
+>;
+type SetAggParamValue<T extends AggParams = any> = ActionType<
```

Comment:
I think that we can replace `any` to a real value:
```suggestion
type SetAggParamValue<T extends AggParams = AggParams> = ActionType<
```

Reply:
Done